### PR TITLE
CHE-137. Improve full text searching

### PR DESCRIPTION
**Pull request by RomanNikitenko**
1. Escape those characters that QueryParser expects to be escaped to correct search
2. Change behavior of full text searching: 
   - old mode: on server side some text for searching splited on separate words and you get all files which contain at least one of the word 
   - new mode - we will find only text for searching as particular sequence of words
3. Change UI for 'Find text' window:
   before my changes
   search_text_before
   after my changes

search_text_after

@skabashnyuk @vparfonov @slemeur

### CODE REVIEW

**Comment by skabashnyuk** [comment]
need some test for server side

**Comment by skabashnyuk** [comment]
-1 
I think we can stay on one query parser.
Issue need some more investigation

File: core/ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/search/FullTextSearchViewImpl.ui.xml [hunk 1]
```diff
@@ -23,80 +23,95 @@
...
+            float: left;
             box-shadow: 0 0 0 1px buttonHoverBorderColor, 0 0 0 2px buttonHoverBackground;
             border-color: transparent;
         }
 
-        .floating {
+        .section {
             float: left;
+            line-height: 22px;
+            margin: 0 0 0 4px;
```
Comment (vzhukovs) [inline core/ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/search/FullTextSearchViewImpl.ui.xml]: margin-left: 4px;

Comment (RomanNikitenko) [inline core/ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/search/FullTextSearchViewImpl.ui.xml]: thank you, Vlad. I'll fix.

**Comment by RomanNikitenko** [comment]
@skabashnyuk @vparfonov I've updated pull request, please review.

File: wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/vfs/search/impl/LuceneSearcher.java [hunk 2]
```diff
@@ -242,15 +242,15 @@ private Query createLuceneQuery(QueryExpression query) throws ServerException {
         final String path = query.getPath();
         final String text = query.getText();
         if (path != null) {
-            luceneQuery.add(new PrefixQuery(new Term("path", path)), BooleanClause.Occur.MUST);
+            luceneQuery.add(new PrefixQuery(new Term("path", path)), MUST);
```
Comment (skabashnyuk) [inline wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/vfs/search/impl/LuceneSearcher.java]: I think we should keep BooleanClause.Occur.MUST way. It has much more descriptive.

**Comment by skabashnyuk** [comment]
other ok

File: wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/vfs/search/impl/LuceneSearcher.java (hunk 2 above)
Comment (RomanNikitenko) [inline wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/vfs/search/impl/LuceneSearcher.java]: @skabashnyuk done!

**Comment by slemeur** [comment]
OK for me

**Comment by vparfonov** [comment]
ok

**Pull request merged**